CDCgov/prime-reportstream
# Sender files - add parameter for message ID

This PR adds a parameter for MessageID. Often times a receiver will request to a single message resent rather than an entire file. Previously the sender-files tool only accepted receiver report IDs and retrieved all messages in that file. This update allow retrieval of a single message.

New parameter: `message-id`

Test Steps:
1. package project locally
2. docker-compose up
3. run `./gradlew resetDB`
4. run `./gradlew reloadTables`
5. run `./gradlew reloadSettings`
6. run command to send test file into docker
     ` curl -X POST -H 'client: simple_report' -H 'Content-Type: text/csv' --data-binary 'PATH/TO/FILE/sr-test-file.csv' 'http://localhost:7071/api/reports'`
     **Test file:** [sr-test-file.csv](https://github.com/CDCgov/prime-reportstream/files/8672820/sr-test-file.csv)

7. Check local covid_results_metadata table to ensure both messages from test file were saved. **There should only be two records in table**
8. Copy messageID from one of the results in covid_results_metatdata. Should be one of these values: **544687, 761936**
9. Run `./prime sender-files --message-id <MESSAGE_ID> -o <PATH/TO/OUTPUT/FOLDER>`
10. Check file in output folder to ensure only the message with the message id you selected is present. **There should only be one message in the file**

## Changes
- Added query to database access to check metadata table
- Added message id parameter to SenderFilesCommand
- Added processing for message id in checkParameters and processRequest in SenderFilesFunction
-  Added test for checkParameters and processRequest to SenderFilesFunctionTests

## Checklist

### Testing
- [] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [X] Test for messageID in checkParameters in SenderFilesFunctionTests
- [X] Test for messageID in processRequest in SenderFilesFunctionTests

### Process

## To Be Done
- Extend functionality to accept csv of messageIDs
